pageserver/src/tenant/storage_layer.rs

```diff
@@ -83,6 +84,11 @@ pub enum ValueReconstructResult {
 /// Supertrait of the [`Layer`] trait that captures the bare minimum interface
 /// required by [`LayerMap`].
 pub trait Layer: Send + Sync {
```

Comment:
A bit unrelated to the PR (feel free to ignore), but some thoughts on the direction of this code:

Judging by the trait docstring, this trait is to `LayerMap` what `Hash` is to `HashMap`. The trait is about key types that are good enough to be indexed by the layer map.

But the name `Layer` implies something more, and talks about the value contained too. The method `get_value_reconstruct_data` is definitely not in the `bare minimum interface required by LayerMap`, but it is a common method that all layers should implement, so it's here.

A possible fix is to rename this trait to `LayerRange` (or `LayerMetadata`, or `LayerDescriptor`), and add a separate trait for `LayerContent`, containing `get_value_reconstruct_data` and `dump`. After this change we won't have anything named `Layer`, which is IMO a good thing. If there was such a thing it should be an enum, not a trait. Or if it is a trait it should just be `pub trait Layer: LayerRange + LayerContent {}`.

Then the only outliers remaining are `get_tenant_id` and `get_timeline_id`. I don't know why these are necessary. The owner of the layer is a timeline, so whoever is using the layer already has a way to get the tenant and timeline id. Maybe they're here just for convenience? Do they have to be part of the trait?